pkg/handler/collector/git/git_test.go

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git_collector
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+func Test_gitCol_RetrieveArtifacts(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
+	logger := logging.FromContext(ctx)
+	type fields struct {
+		url      string
+		dir      string
+		poll     bool
+		interval time.Duration
+	}
+	tests := []struct {
+		name                   string
+		fields                 fields
+		preCreateDir           bool
+		wantErr                bool
+		numberOfFilesCollected int
+	}{{
+		name: "get repo",
+		fields: fields{
+			url:      "https://github.com/shafeeshafee/test-123.git",
```

Comment:
@lumjjb @pxp928 We should probably pull this test repo into guacsec itself. Otherwise we end up in a situation where a contributor could modify this repo and break the tests or potentially cause malicious behavior.

Thoughts?

Reply:
Yes I agree. For the unit test wanted a simple repo that proved the functionality without much effort. We should make that repo private so it's not visible to others.

Reply:
Changed to https://github.com/guacsec/git-collector-test